stan/math/prim/fun/to_int.hpp

```diff
@@ -0,0 +1,59 @@
+#ifndef STAN_MATH_PRIM_FUN_TO_INT_HPP
+#define STAN_MATH_PRIM_FUN_TO_INT_HPP
+
+#include <stan/math/prim/functor/apply_scalar_unary.hpp>
+
+namespace stan {
+namespace math {
+
+/**
+ * Returns the input scalar as an integer type
```

Comment:
This needs to be much more specific in terms of how it returns.  You can say it truncates using `static_cast<int>` to be precise, but something needs to say how it converts.